core/io/xml_parser.cpp

```diff
@@ -32,12 +32,18 @@
 
 #include "core/string/print_string.h"
 
+#include <ctype.h>
+
 //#define DEBUG_XML
 
 static inline bool _is_white_space(char c) {
 	return (c == ' ' || c == '\t' || c == '\n' || c == '\r');
 }
 
+static inline bool _is_punctuation(char c) {
+	return !((int(c) >= 65 && int(c) <= 90) || (int(c) >= 97 && int(c) <= 122));
```

Comment:
Not sure if the case (int) is required here.

However the logic isn't correct based off 
https://www.w3schools.com/xml/xml_elements.asp
XML Naming Rules
Element names can contain letters, digits, hyphens, underscores, and periods.

Reply:
But tags can't [https://stackoverflow.com/questions/19677315/xml-tagname-starting-with-number-is-not-working](https://stackoverflow.com/questions/19677315/xml-tagname-starting-with-number-is-not-working) , and that is what that method is for

Reply:
Sorry that was my misunderstand.

Reply:
> Not sure if the case (int) is required here.

What do you mean by that? I'm taking advantage of the fact that each char has a corresponding number in the ascii table, hence I can check if a char is a letter by comparing its number to the letters (A->Z, a->z) number in the aforementioned table

Reply:
Why not just write it like this, which IMO is clearer as to what your trying to achieve
!( (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c == '_'));

Also I believe an underscore is a valid first character

Reply:
> Why not just write it like this, which IMO is clearer as to what your trying to achieve !( (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c == '_'));

Yeah it's equivalent I guess, let's see what others say about it and then in case I'll change it :D

> 
> Also I believe an underscore is a valid first character

It is not. You can check in any online xml viewer/formatter by providing a small example

Reply:
The following validates with online XML checkers.
```
<XMI>
<_Name xxx="c" />
</XMI>
```

Reply:
> The following validates with online XML checkers.
> 
> ```
> <XMI>
> <_Name xxx="c" />
> </XMI>
> ```

Yep, I got confused ahah in fact I edited my previous answer, sorry about that